Allow Pseudo TAs to define their TA property flags

etienne-lms opened the pull request:
As part of the Secure Data Path, pseudo TAs may need to access SDP memory buffers using the memory reference provided as invocation parameters by their client.
As most pseudo TAs are likely to be not much interested with the SDP feature, these change aims at allowing pseudo TA to declare to OP-TEE core whether or not they do handle SDP memory references (and the several related constraints). Only such TAs will be able to be invoked with SDP memref parameters.

This PR does not talk about SDP. It proposes pseduo TA features that will be required for the SDP support intergration (https://github.com/OP-TEE/optee_os/pull/1322).

jenswi-linaro commented on core/arch/arm/pta/se_api_self_tests.c:
```diff
@@ -77,30 +77,6 @@
  * Trusted Application Entry Points
  */
 
-static TEE_Result create_ta(void)
```
I could be useful to keep these for this TA to avoid bitrot.

etienne-lms replied:
I don't understand the issue ?
Does your comment apply to all pTAs, or only to this SE API test setup?

jenswi-linaro replied:
Sorry, I thought I was commenting in `core/arch/arm/pta/pta_self_tests.c`. Just keeping it in `core/arch/arm/pta/pta_self_tests.c` should be enough.

etienne-lms replied:
Ah, ok, i understand. Ok, I keep the create_ta() handler for the self test pTA with its trace.

jenswi-linaro commented:
With or without my comment addressed:
`Reviewed-by: Jens Wiklander <jens.wiklander@linaro.org>`

jforissier commented on core/arch/arm/kernel/pseudo_ta.c:
```diff
@@ -173,6 +175,39 @@ static const struct tee_ta_ops pseudo_ta_ops = {
 extern const struct pseudo_ta_head __start_ta_head_section;
 extern const struct pseudo_ta_head __stop_ta_head_section;
 
+/* Insures declared pseudo TAs confirms with core expectations */
```
conforms

etienne-lms replied:
thanks

etienne-lms commented:
updated, rebased, tags applied (I anticipated your review comments, since you both seemed to agree with the goal of these changes! @jforissier feel free to reject this if necessary).

jforissier commented:
@etienne-lms it's OK, thanks!
